tests/drivers/gpio/gpio_basic_api/testcase.yaml

```diff
@@ -36,20 +36,66 @@ tests:
     depends_on:
       - gpio
       - arduino_gpio
-    filter: dt_compat_enabled("test-gpio-basic-api") and dt_compat_enabled("arduino-header-r3")
+    filter: >
+      dt_compat_enabled("test-gpio-basic-api") and dt_compat_enabled("arduino-header-r3")
+      and not CONFIG_SOC_FAMILY_STM32
     extra_args: "DTC_OVERLAY_FILE=arduino.overlay"
+    extra_configs:
+      - CONFIG_ONE_PORT=n
```

Comment:
It seems wrong that `drivers.gpio.2pin_arduino:` which looks as the default test is set with `CONFIG_ONE_PORT=n` which is not default test configuration